arangod/RestServer/DatabaseFeature.h

```diff
@@ -23,25 +23,42 @@
 #ifndef APPLICATION_FEATURES_DATABASE_FEATURE_H
 #define APPLICATION_FEATURES_DATABASE_FEATURE_H 1
 
+#if !defined(USE_CATCH_TESTS) && !defined(EXPAND_APPLICATION_FEATURES_DATABASE_FEATURE_H)
+  #define DO_EXPAND_APPLICATION_FEATURES_DATABASE_FEATURE_H(VAL) VAL ## 1
+  #define EXPAND_APPLICATION_FEATURES_DATABASE_FEATURE_H(VAL) DO_EXPAND_APPLICATION_FEATURES_DATABASE_FEATURE_H(VAL)
+  #if defined(TEST_VIRTUAL) && (EXPAND_APPLICATION_FEATURES_DATABASE_FEATURE_H(TEST_VIRTUAL) != 1)
+    #define USE_CATCH_TESTS
+  #endif
+  #undef EXPAND_APPLICATION_FEATURES_DATABASE_FEATURE_H
+  #undef DO_EXPAND_APPLICATION_FEATURES_DATABASE_FEATURE_H
+#endif
```

Comment:
can you _please_ fix the CMakeList.txt instead of this ?!

Reply:
The CMakeLists.txt does not have any issues to my knowledge. It defines TEST_VIRTUAL=virtual if catch tests are enabled.
This code was copied from FlushFeature.h, it's intent is to define a macro if TEST_VIRTUAL is set to any value v.s. empty or absent.
The only other approach I can think of is to add more defines in CMakeLists.txt instead of reusing existing ones, but I am not sure if that is desirable.

Reply:
how about adding `add_definitions("-DUSE_CATCH_TESTS=1")` to CMakeList.txt

Reply:
or `ARANGODB_CATCH_TESTS` to avoid collisions

Reply:
that's exactly what I meant by "The only other approach I can think of is to add more defines in CMakeLists.txt instead of reusing existing ones, but I am not sure if that is desirable."
The question still remains, something that should probably be discussed with at least @dothebart and @jsteemann